### PR TITLE
WT-12388 Move on to the next collection when background compaction conflicts with a running checkpoint

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/session/session_compact.c [hunk 1]
```diff
@@ -289,7 +289,9 @@ __compact_worker(WT_SESSION_IMPL *session)
 {
     WT_DECL_RET;
     u_int i, loop;
-    bool another_pass;
+    bool another_pass, background_compaction;
```
Comment (keitharnoldsmith) [inline src/session/session_compact.c:292]: Alphabetize, please.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.

One minor nit, but it doesn't need me to re-review.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by Sean04 — APPROVED** [review]
LGTM

**Pull request merged**